elements/lisk-p2p/src/peer_book/base_list.ts

```diff
@@ -121,66 +128,25 @@ export class PeerList {
 		return false;
 	}
 
-	public getPeer(peerInfo: P2PPeerInfo): P2PPeerInfo | undefined {
-		const bucketId = this.selectBucketId(peerInfo.ipAddress);
-		const bucket = this.peerMap.get(bucketId);
-		const incomingPeerId = constructPeerIdFromPeerInfo(peerInfo);
-
-		if (!bucket) {
-			return undefined;
-		}
-		const peer = bucket.get(incomingPeerId);
+	public getBucket(ipAddress: string): Map<string, CustomPeerInfo> {
+		const bucketId = getBucketId({
+			secret: this.peerListConfig.secret,
+			peerType: this.peerListConfig.peerType,
+			targetAddress: ipAddress,
+			bucketCount: this.peerListConfig.peerBucketCount,
+		});
 
-		return peer ? peer.peerInfo : undefined;
+		return this.peerMap.get(bucketId) as Map<string, CustomPeerInfo>;
 	}
 
-	public initPeerInfo = (peerInfo: P2PPeerInfo): CustomPeerInfo => ({
-		peerInfo,
-		dateAdded: new Date(),
-	});
-
-	// Addition of peer can also result in peer eviction if the bucket of the incoming peer is already full based on evection strategy.
-	public addPeer(peerInfo: P2PPeerInfo): AddPeerOutcome {
-		const bucketId = this.selectBucketId(peerInfo.ipAddress);
-		const bucket = this.peerMap.get(bucketId);
-		const incomingPeerId = constructPeerIdFromPeerInfo(peerInfo);
-
-		if (!bucket) {
-			return {
-				success: false,
-				isAdded: false,
-				evictedPeer: undefined,
-			};
-		}
-
-		if (bucket && bucket.get(incomingPeerId)) {
-			return {
-				success: false,
-				isAdded: false,
-				evictedPeer: undefined,
-			};
-		}
-
-		const newPeer = this.initPeerInfo(peerInfo);
-
-		if (bucket.size < this.peerListConfig.peerBucketSize) {
-			bucket.set(incomingPeerId, newPeer);
+	public makeSpace(ipAddress: string): CustomPeerInfo | undefined {
```

Comment:
why it's not `addPeer` anymore? if the space is empty and just simply adding peer, would it use this function?

Reply:
`makeSpace()` checks if the bucket is full and only in that case removes a peer using eviction strategies and returns that evicted peer. If the bucket is not full, it just returns undefined without removing any peer. After that, `addPeer()` is executed.

Reply:
I am not sure about the naming. maybe `checkBucket`

Reply:
At the end, all of them are incomplete. MakeSpace is simply and we just need to write in the header that it does it if there is not space.

Reply:
Yes, let's keep it simple as of now